packages/token-bridge-sdk/src/hooks/useGasPrice.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useMemo, useState } from 'react'
+import { BigNumber, providers } from 'ethers'
+import useSWR from 'swr'
+
+async function tryFetchGasPrice({
+  provider
+}: {
+  provider: providers.Provider
+}) {
+  try {
+    return await provider.getGasPrice()
+  } catch (error) {
+    return BigNumber.from(0)
+  }
+}
+
+export function useGasPrice({
+  provider
+}: {
+  provider: providers.Provider
+}): BigNumber {
+  const [chainId, setChainId] = useState<number | null>(null)
+
+  useEffect(() => {
+    async function updateChainId() {
+      setChainId((await provider.getNetwork()).chainId)
+    }
+
+    updateChainId()
```

Comment:
I might miss the reason, but can we call directly `setChainId((await provider.getNetwork()).chainId)` without creating a new function?

Reply:
Yep - effects can't be async, and I didn't wanna do an IIFE